### ISSUE TITLE
Feature Request: Add Ollama-compatible APIs

### ISSUE THREAD

**Issue by arnesund** [gpustack/gpustack#299]
I'm setting up Open WebUI as the web frontend for internal use of LLMs, complete with HA + SSO setup and all. GPUStack is used as an OpenAI-compatible backend and it works great for web-based access. 

I'm also exploring ways to make it easy for our devs to get API access to the same wide variety of LLMs. Open WebUI exposes an /ollama API endpoint to passthrough requests to the Ollama-compatible backends, after authenticating requests. It doesn't support this for OpenAI-compatible backends. Any chance you'll consider to add Ollama-compatible APIs to GPUStack?

That would make this use-case easier to realize, by still having only one entrypoint (Open WebUI) where authentication is handled before passthrough of requests to GPUStack.

**Comment by gitlawr** [comment]
OpenAI-compatible APIs are widely adopted in modern inference engines(llama.cpp, vllm, sglang, etc.) and local serving tools(lm studio, localai, etc.). I think it makes more sense if existing authentication in Open WebUI can work with any OpenAI-compatible backends.

**Comment by gitlawr** [comment]
We have multiple inference backends in the up-coming 0.3 release. If Ollama APIs are convient for users we may consider running ollama models in ollama-native way(another backend) so that we can serve Ollama APIs besides OpenAI APIs.

**Comment by arnesund** [comment]
> I think it makes more sense if existing authentication in Open WebUI can work with any OpenAI-compatible backends.

Agree. And I discovered now that I was wrong about Open WebUI not supporting passthrough to OpenAI-compatible API backends, they actually have that feature: https://docs.openwebui.com/getting-started/api#swagger-documentation-links
I tested it and it works! I can send an API request to the Open WebUI service, with an API key created myself in that UI, and it gets routed to GPUStack and to the right model. Therefore I'll close this feature request.

Great to hear that you'll support multiple inference backends too, in the upcoming release. Keep up the great work!
